### PR TITLE
Ignore HTTP_PROXY in reverse tunnels, part 2

### PR DESCRIPTION
**Pull request by atburke**
This PR disables HTTP_PROXY in a few places that were missed in #11990.

### CODE REVIEW

File: lib/auth/clt.go [hunk 1]
```diff
@@ -129,7 +129,8 @@ func NewHTTPClient(cfg client.Config, tls *tls.Config, params ...roundtrip.Clien
 		if len(cfg.Addrs) == 0 {
 			return nil, trace.BadParameter("no addresses to dial")
 		}
-		contextDialer := client.NewDialer(cfg.KeepAlivePeriod, cfg.DialTimeout)
+		// Deliberately ignore HTTP proxies for backwards compatibility.
+		contextDialer := client.NewDirectDialer(cfg.KeepAlivePeriod, cfg.DialTimeout)
```
Comment (r0mant) [inline lib/auth/clt.go:133]: Is this change related to the reverse tunnel issue? Won't this break the actual scenario in which tsh should be respecting the proxies?

File: lib/reversetunnel/resolver.go [hunk 2]
```diff
@@ -59,7 +59,7 @@ func WebClientResolver(ctx context.Context, addrs []utils.NetAddr, insecureTLS b
 			// In insecure mode, any certificate is accepted. In secure mode the hosts
 			// CAs are used to validate the certificate on the proxy.
 			tunnelAddr, err := webclient.GetTunnelAddr(
-				&webclient.Config{Context: ctx, ProxyAddr: addr.String(), Insecure: insecureTLS})
+				&webclient.Config{Context: ctx, ProxyAddr: addr.String(), Insecure: insecureTLS, IgnoreHTTPProxy: true})
```
Comment (r0mant) [inline lib/reversetunnel/resolver.go:62]: Add a comment explaining why we explicitly ignore it here.

**Review by r0mant — COMMENTED** [review]
Let's please also add test coverage to make sure it does not regress again.

**Comment by atburke** [comment]
@r0mant Can you take another look at this?

File: lib/auth/authclient/authclient.go [hunk 3]
```diff
@@ -55,6 +55,8 @@ func Connect(ctx context.Context, cfg *Config) (auth.ClientI, error) {
 		Credentials: []apiclient.Credentials{
 			apiclient.LoadTLS(cfg.TLS),
 		},
+		// Deliberately ignore HTTP proxies for backwards compatibility.
+		IgnoreHTTPProxy: true,
```
Comment (r0mant) [inline lib/auth/authclient/authclient.go:59]: I'm probably missing something, but I'm still not sure why we need to ignore it in the authclient too. Isn't lib/reversetunnel/resolver enough?

**Review by r0mant — COMMENTED** [review]
@atburke The code lgtm in general but I'm having a bit of a hard time understanding in which cases we'll be respecting vs not respecting proxies with these changes. Mostly because these clients are used throughout the code.

Before we merge this, could you please help me clarify my understanding regarding the following:

- In which scenarios the proxy will be respected? When using tsh only?
- In which scenarios the proxy will not be respected? For inter-process communication?
- In which scenarios the proxy was respected prior to your original HTTP_PROXY changes? Never?

**Comment by atburke** [comment]
Previously, HTTP_PROXY was only supported (as far as I can tell) for inter-process communication in single-port mode. My original HTTP_PROXY PR added support deliberately for `tsh` and accidentally for IPC in multi-port mode. This PR removes support for IPC in multi-port mode. The end result should be that HTTP_PROXY is supported for `tsh` and multi-port mode, and not in single-port mode.

File: lib/auth/authclient/authclient.go (hunk 3 above)
Comment (atburke) [reply lib/auth/authclient/authclient.go:59]: It isn't enough. authclient was also affected by the original HTTP_PROXY change. `TestMultiPortNoProxy` doesn't pass unless authclient ignores proxies.

**Review by r0mant — APPROVED** [review]

**Review by nklaassen — APPROVED** [review]
Looks alright to me.

Do we have any docs on when exactly this env variable applies? I'm somewhat confused, I imagine it could be pretty confusing for users

**Review by gabrielcorado — APPROVED** [review]

**Comment by atburke** [comment]
We have a [docs page for http proxy](https://goteleport.com/docs/setup/reference/networking/#http-connect-proxies). I'll update it with the current behavior in a new PR.
